cmd/posix.go

```diff
@@ -367,30 +364,36 @@ func (s *posix) getVolDir(volume string) (string, error) {
 	return volumeDir, nil
 }
 
-// checkDiskFound - validates if disk is available,
-// returns errDiskNotFound if not found.
-func (s *posix) checkDiskFound() (err error) {
-	if !s.IsOnline() {
-		return errDiskNotFound
+func (s *posix) getDiskID() (string, error) {
+	s.RLock()
+	diskID := s.diskID
+	s.RUnlock()
+
+	formatFile := pathJoin(s.diskPath, minioMetaBucket, formatConfigFile)
+	fi, err := os.Stat(formatFile)
+	if err != nil {
+		// If the disk is still not initialized.
+		return "", err
+	}
+
+	if os.SameFile(fi, s.formatFileInfo) && fi.ModTime() == s.formatFileInfo.ModTime() {
+		// If the file has not changed, just return the cached diskID information.
+		return diskID, nil
 	}
-	fi, err := os.Stat(s.diskPath)
+
+	s.Lock()
+	defer s.Unlock()
+	b, err := ioutil.ReadFile(formatFile)
```

Comment:
You can avoid the os.Stat() and rely on ioutil.ReadFile() instead to indicate if format.json exists?

Reply:
actually, I am using `os.Stat()` to avoid `ioutil.ReadFile()` as `ReadFile()` is much more expensive

Reply:
os.Open() v/s os.Stat() + os.Open() - we can avoid the second call because - for the most part there is formatfile @krishnasrinivas - you are just taxing the good case more than the rare case